sdk/keyvault/azure-security-keyvault-common/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required (VERSION 3.13)
+project(azure-security-keyvault-common LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake-modules")
+
+include(az_version)
+include(CodeCoverage)
+include(DefineTransportAdapter)
+include(doxygen_common)
+include(global_compile_options)
+
+add_library(
+    azure-security-keyvault-common
+    src/keyvault_exception.cpp
+    src/keyvault_pipeline.cpp
+)
+
+target_include_directories (azure-security-keyvault-common PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc> $<INSTALL_INTERFACE:include/azure-security-keyvault-common>)
```

Comment:
I was hoping not to add the vcpkg support for keyvault since now.

Initially, I just want the keyvault project to be there and we can either build it from root or individually. But not yet to be vcpkg until its first preview maybe.

Reply:
We don't have to release it as vcpkg (on the other hand, why not?), but I think it is better to have it vcpkg-ready. Are we going to release it one day?
Example: we have azure-template vcpkg-ready, never even going to release it.